imap_processing/tests/ena_maps/test_ena_maps.py

```diff
@@ -192,23 +192,76 @@ def test_project_pset_values_to_map_push_method(self, mock_frame_transform_az_el
             simple_summed_pset_counts,
         )
 
+    @pytest.mark.usefixtures("_setup_ultra_l1c_pset_products")
+    def test_project_pset_values_to_map_errors(self):
+        index_matching_method = ena_maps.IndexMatchMethod.PUSH
+        rectangular_map = ena_maps.RectangularSkyMap(
+            spacing_deg=1,
+            spice_frame=geometry.SpiceFrame.ECLIPJ2000,
+        )
+
+        # An error should be raised if a key is not found in the PSET
+        with pytest.raises(ValueError, match="Value key invalid not found"):
+            rectangular_map.project_pset_values_to_map(
+                self.ultra_psets[0],
+                value_keys=["invalid"],
+                index_match_method=index_matching_method,
+            )
+
     @pytest.mark.usefixtures("_setup_ultra_l1c_pset_products")
     @mock.patch("imap_processing.spice.geometry.frame_transform_az_el")
     def test_project_pset_values_to_map_pull_method(self, mock_frame_transform_az_el):
         """Test projection to Rect. Map fails w "pull" index matching method."""
 
         index_matching_method = ena_maps.IndexMatchMethod.PULL
+        skymap_spacing = 10
+
+        # Mock frame_transform to return the az and el unchanged
+        mock_frame_transform_az_el.side_effect = (
+            lambda et, az_el, from_frame, to_frame, degrees: az_el
+        )
         rectangular_map = ena_maps.RectangularSkyMap(
-            spacing_deg=10,
+            spacing_deg=skymap_spacing,
             spice_frame=geometry.SpiceFrame.ECLIPJ2000,
         )
 
-        with pytest.raises(NotImplementedError):
+        # Each map pixel will add the value of a single PSET pixel, so we'll start at 0
+        # and add 0, 1, 2, 3, ... to the map
+        expected_value_every_pixel = 0
+
+        # Another way to test this is that (if the PSET pixels are
+        # smaller than the SkyMap pixels) the sum of the counts in all PSETs should
+        # be (PSET_spacing / SkyMap_spacing)^2 times the sum of the counts in the SkyMap
+        total_pset_counts = np.zeros_like(self.ultra_psets[0].data["counts"].values)
+
+        # Project each PSET's values to the map (pull method)
+        for pset_num, ultra_pset in enumerate(self.ultra_psets):
+            # Set the counts to be 0 in the first PSET, 1 in the second, etc.
+            ultra_pset.data["counts"].values = np.full_like(
+                ultra_pset.data["counts"].values, pset_num
+            )
+
             rectangular_map.project_pset_values_to_map(
-                self.ultra_psets[0],
+                ultra_pset,
                 value_keys=["counts", "exposure_time"],
                 index_match_method=index_matching_method,
             )
+            expected_value_every_pixel += pset_num
+
+            total_pset_counts += ultra_pset.data["counts"].values
+
+        # Check that the map has been updated
+        assert rectangular_map.data_dict != {}
```

Comment:
assert some key is in it instead? This just says it isn't empty but not what you expect to be in it.

Reply:
Switching to:
```assert "counts" in rectangular_map.data_dict```